app/models/compute_resources/foreman/model/gce.rb

```diff
@@ -60,20 +63,52 @@ def zone=(zone)
       self.url = zone
     end
 
-    def create_vm(args = {})
-      #Dot are not allowed in names
+    def new_vm(args = {})
+      # convert rails nested_attributes into a plain hash
+      [:volumes].each do |collection|
+        nested_attrs = args.delete("#{collection}_attributes".to_sym)
+        args[collection] = nested_attributes_for(collection, nested_attrs) if nested_attrs
+      end
+
+      # Dots are not allowed in names
       args[:name]        = args[:name].parameterize if args[:name].present?
-      args[:external_ip] = args[:external_ip] != '0'
-      args[:image_name]  = args[:image_id]
+      args[:external_ip] = args[:external_ip] == '1'
+      # GCE network interfaces cannot be defined though Foreman yet
+      args[:network_interfaces] = nil
+
+      if args[:volumes].present?
+        if args[:image_id].present?
+          args[:volumes].first[:source_image] = client.images.select { |i| i.id == args[:image_id] }.first.name
+        end
+        args[:disks] = []
+        args[:volumes].each_with_index do |vol_args,i|
+          args[:disks] << new_volume(vol_args.merge(:name => "#{args[:name]}-disk#{i+1}"))
+        end
+      end
+
+      super(args)
+    end
+
+    def create_vm(args = {})
+      new_vm(args)
+      create_volumes(args)
 
       username = images.where(:uuid => args[:image_name]).first.try(:username)
       ssh      = { :username => username, :public_key => key_pair.public }
-      super(args.merge(ssh))
+      vm = super(args.merge(ssh))
+      vm.disks.each { |disk| vm.set_disk_auto_delete(true, disk['deviceName']) }
```

Comment:
I've updated this line, it was using `args[:disks]` names, but it turns out `vm.disks` uses a slightly different naming. This prevents leaving orphaned disks after an instance is deleted.